pandas/core/frame.py

```diff
@@ -3152,6 +3152,13 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
         else:
             from pandas.core.groupby import _nargsort
 
+            if (ascending and labels.is_monotonic_increasing) or \
```

Comment:
pls add a comment describing what you are doing here and a reference to the issue